### PR TITLE
[#174006044] Log users logins

### PR DESCRIPTION
**Pull request by francescopersico**

### CODE REVIEW

File: src/services/usersLoginNotificationService.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+
+export const UserLogin = t.interface({
+  fiscalCode: FiscalCode,
+  lastLoginAt: UTCISODateFromString,
+  source: t.keyof({
+    cie: null,
+    spid: null
+  })
+});
```
Comment (BurnedMarshal) [inline src/services/usersLoginNotificationService.ts:15]: Can we use something like this? IMHO is more clear.
```suggestion
const sourceType = t.union([t.literal("spid"), t.literal("cie")]);
type sourceType = t.TypeOf<typeof sourceType>;

export const UserLogin = t.interface({
  fiscalCode: FiscalCode,
  lastLoginAt: UTCISODateFromString,
  source: sourceType
});
```

Comment (francescopersico) [reply src/services/usersLoginNotificationService.ts:15]: @BurnedMarshal  You mean to extract the source type, to use "union" instead of "keyof", or both?

Comment (BurnedMarshal) [reply src/services/usersLoginNotificationService.ts:15]: "union" instead of "keyof" could be enough.

Comment (francescopersico) [reply src/services/usersLoginNotificationService.ts:15]: @BurnedMarshal https://github.com/gcanti/io-ts/tree/1.x#union-of-string-literals

File: src/services/usersLoginNotificationService.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      this.queueName
+    );
+  }
+
+  public readonly notifyUserLogin = (
+    userLogin: UserLogin
+  ): Promise<QueueSendMessageResponse> => {
+    const userLoginEncoded = UserLogin.encode(userLogin);
+
+    return this.queueClient.sendMessage(JSON.stringify(userLoginEncoded));
```
Comment (gunzip) [inline src/services/usersLoginNotificationService.ts:37]: shouldn't this be base64 encoded?

File: .env.example [hunk 3]
```diff
@@ -32,5 +32,7 @@ SPID_LOG_STORAGE_CONNECTION_STRING=DefaultEndpointsProtocol=http;AccountName=dev
 SPID_LOG_QUEUE_NAME=spidqueuename
 NOTIFICATIONS_STORAGE_CONNECTION_STRING=foobar
 NOTIFICATIONS_QUEUE_NAME=nqueuename
+USERS_LOGIN_STORAGE_CONNECTION_STRING=foobar
+USERS_LOGIN_QUEUE_NAME=ulqueuename
```
Comment (BurnedMarshal) [inline .env.example:36]: Could you add these variables into the README

Comment (francescopersico) [reply .env.example:36]: Done

File: src/services/usersLoginNotificationService.ts (hunk 2 above)
Comment (francescopersico) [reply src/services/usersLoginNotificationService.ts]: Done

File: src/app.ts [hunk 4]
```diff
@@ -74,6 +76,7 @@ import RedisSessionStorage from "./services/redisSessionStorage";
 import RedisUserMetadataStorage from "./services/redisUserMetadataStorage";
 import TokenService from "./services/tokenService";
 import UserDataProcessingService from "./services/userDataProcessingService";
+import UsersLoginNotificationService from "./services/usersLoginNotificationService";
```
[the remark this answers is not in the document]
Comment (francescopersico) [reply src/app.ts]: Renamed to UsersLoginLogService

**Pull request merged**